src/linquantarrays.jl

```diff
@@ -10,84 +13,164 @@ Base.size(QA::LinQuantArray) = size(QA.A)
 Base.getindex(QA::LinQuantArray,i...) = getindex(QA.A,i...)
 Base.eltype(Q::LinQuantArray{T,N}) where {T,N} = T
 
-"""Quantise an array linearly into a LinQuantArray."""
-function LinQuantization(::Type{T},A::AbstractArray) where {T<:Unsigned}
+"""
+
+        LinQuantization(::Type{T}, A::AbstractArray; extrema::Tuple = extrema(A)) where {T<:Integer}
+
+Quantise an array linearly into a LinQuantArray.
+
+# Arguments
+- `T`: the type of the quantised array
+- `A`: the array to quantise
+- `extrema`: the minimum and maximum of the range, defaults to `extrema(A)`.
+
+# Returns
+- a LinQuantArray{T} with the quantised array and the minimum and maximum of the original range.
+"""
+
+function LinQuantization(
+    ::Type{T},
+    A::AbstractArray,
+    extrema::Nothing
+) where {T<:Integer}
     all(isfinite.(A)) || throw(DomainError("Linear quantization only in (-∞,∞)"))
 
-    Amin = Float64(minimum(A))              # minimum of value range
-    Amax = Float64(maximum(A))              # maximum of value range
+    # range of values in A
+    Amin, Amax  = Float64(minimum(A)), Float64(maximum(A))    
```

Comment:
I would use `extrema` from `Base` here. Otherwise the array `A` has to be scanned twice, `extrema` does both in one -- important for very large arrays and performance